src/main/java/teammates/ui/webapi/action/GetCourseAction.java

```diff
@@ -19,22 +19,26 @@ protected AuthType getMinAuthLevel() {
 
     @Override
     public void checkSpecificAccessControl() {
-        if (!(userInfo.isInstructor || userInfo.isStudent)) {
-            throw new UnauthorizedAccessException("Student or instructor account is required to access this resource.");
-        }
-
         String courseId = getNonNullRequestParamValue(Const.ParamsNames.COURSE_ID);
 
-        if (userInfo.isInstructor) {
+        boolean checkForInstructorAccess = logic.getInstructorForGoogleId(courseId, userInfo.getId()) != null;
```

Comment:
Consider using `Const.ParamsNames.ENTITY_TYPE)`? What do you think?

I don't really like the idea of identifying "who you are" in the server side. It is expensive to do that. Instead, I would prefer the client (the frontend) to explicitly say the intent (e.g., indicate whether you are requesting the resource on behalf of student or instructor).